tests/models/test_match_spec.py

```diff
@@ -642,3 +660,47 @@ def test_parse_parens(self):
             # "target": "blarg",  # suppressing these for now
             # "optional": True,
         }
+
+    def test_parse_build_number_brackets(self):
+        assert _parse_spec_str("python[build_number=3]") == {
+            "name": "python",
+            "build_number": '3',
+        }
+        assert _parse_spec_str("python[build_number='>3']") == {
+            "name": "python",
+            "build_number": '>3',
+        }
+        assert _parse_spec_str("python[build_number='>=3']") == {
+            "name": "python",
+            "build_number": '>=3',
+        }
+
+        assert _parse_spec_str("python[build_number='<3']") == {
+            "name": "python",
+            "build_number": '<3',
+        }
+        assert _parse_spec_str("python[build_number='<=3']") == {
+            "name": "python",
+            "build_number": '<=3',
+        }
+
+        # # these don't work right now, should they?
```

Comment:
It would be nice to support these un-quoted but I wouldn't make them a high priority.